Tweaked zombie soldier, cop and survivor drop lists

Night-Pryanik opened the pull request:
#### Summary
```SUMMARY: Balance "Tweaked zombie soldier, cop and survivor drop lists."```

#### Purpose of change
Zombie soldiers have too generous loot table. Long-barreled weapons are held in hands of zombie soldiers, cops, survivors, and ordinary zombies (and they shouldn't).

#### Describe the solution
1. Every zombie soldier _always_ had weapons on it. I find this unrealistic and unfun. Zombies don't carry weapons in their hands, so one of the was by which they can have weapons is to hang them on their shoulder straps. Having this in mind, I reduced weapon spawn chance from guaranteed to 40%, and added shoulder strap to all these weapons. 
Also added shoulder straps to ordinary zombies, zombie cops and zombie survivors long-barreled firearms.
2. Reduced chance to spawn many soldier gear and clothing from guaranteed to some percent - let's assume they lost it after the zombification (or never have them in the first place).

#### Describe alternatives you've considered
None.

#### Testing
Spawned ~500 zombie soldiers, cops and survivors, debug-killed them all, checked their gear.

#### Additional context
None.

LaVeyanFiend commented:
This shoulder strap addition would be good on all zombie longgun drops, yes?

Night-Pryanik commented:
> This shoulder strap addition would be good on all zombie longgun drops, yes?

Yes. If main firearm spawn chance is rolled, it will spawn with shoulder strap.

LaVeyanFiend commented:
I meant that the addition of the shoulder strap should be added to _all_ zombie longgun drops to fix that issue as well, not just military_standard_X

Night-Pryanik commented:
Ok, I added shoulder straps to zombie cops and zombie survivors long-barreled firearms item groups. AFAIK these are the only ones that are zombie-dropped. If there are other, let me know, I'll update them too.

LaVeyanFiend commented on data/json/monsterdrops/zombie_survivor.json:
```diff
@@ -87,5 +139,30 @@
+      { "item": "shotgun_s", "prob": 10 },
+      { "item": "shotgun_d", "prob": 10, "charges": [ 0, 2 ] },
+      { "item": "sten", "prob": 20, "charges": [ 0, 32 ] },
+      { "item": "sks", "prob": 10, "charges": [ 0, 10 ] }
```
Are the probabiblites supposed to add up to 90 rather than 100?

Night-Pryanik replied:
Probabilities don't sum up here. They just mean that, e.g. `ar15` has four times less chance to spawn than `sten` has.

LaVeyanFiend commented on data/json/monsterdrops/zombie_survivor.json:
```diff
@@ -87,5 +139,30 @@
+      { "item": "remington_870", "prob": 5, "charges": [ 0, 6 ] },
+      { "item": "ruger_1022", "prob": 15, "charges": [ 0, 10 ] },
+      { "item": "shotgun_410", "prob": 10, "charges": [ 0, 1 ] },
+      { "item": "shotgun_s", "prob": 10 },
```
```suggestion
      { "item": "shotgun_s", "prob": 10 , "charges": [ 0, 1 ] },
```

Night-Pryanik commented on data/json/itemgroups/clothing.json:
```diff
@@ -92,18 +92,18 @@
     "items": [
-      { "group": "clothing_soldier_shirt" },
-      { "item": "jacket_army" },
+      { "group": "clothing_soldier_shirt", "prob": 80 },
```
No one knows what happens to this specific soldier after his zombification. He could lost his e.g. army jacket literally anywhere, hence the reduced chance. _Any_ soldier having his army jacket no matter what perils he had to endure is unrealistic and inappropriate.

ymber replied:
It's very unlikely that he's going to have lost basic parts of ACU. Helmet maybe. Not tshirts.

Night-Pryanik commented on data/json/monsterdrops/zombie_soldier.json:
```diff
@@ -14,15 +14,16 @@
+        "prob": 40
       },
-      { "group": "infantry_common_gear" },
+      { "group": "infantry_common_gear", "prob": 30 },
```
We're talking about zombified soldiers, not live ones, here. As I already said, zombified soldiers could (and most probably will) lose at least some of its gear.

kevingranade replied:
I think the pertinent thing here is would they carry them into a firefight? What's doctorine about fighting while carrying a daypack?

Night-Pryanik commented on data/json/monsterdrops/zombie_soldier.json:
```diff
@@ -50,16 +51,18 @@
-          { "group": "military_standard_shotguns", "prob": 5 }
-        ]
+          { "group": "military_standard_lmgs", "contents-item": "shoulder_strap", "prob": 10 },
+          { "group": "military_standard_sniper_rifles", "contents-item": "shoulder_strap", "prob": 10 },
```
Then I'm removing the whole group, as there is no way to hold these weapons without slings.

LaVeyanFiend replied:
Back holsters instead of slings would work

ymber replied:
Back holsters aren't used by infantry. If we're going to assume any weapons zombie soldiers have were on slings removing them from drop lists is actually the reasonable approach. The problem this approach presents is that if they don't spawn on the zombie soldiers they don't spawn anywhere. Their guns shouldn't be evaporating.

Night-Pryanik commented on data/json/monsterdrops/zombie_soldier.json:
```diff
@@ -14,15 +14,16 @@
+      { "group": "infantry_common_gear", "prob": 30 },
       { "group": "military_standard_grenades", "count": [ 1, 3 ], "prob": 20 },
-      { "group": "military_patrol_food" },
+      { "group": "military_patrol_food", "prob": 30 },
```
What are you suggesting?

mkikt4743 replied:
that people carry food onto long-term engagements probably

Amoebka commented:
While the assumpion that zombies can lose some clothing is reasonable, the proposed solution leads to idiotic situations where zombies are missing socks but have boots on.
Inner-layer clothing at least should have 100% to spawn.

Night-Pryanik commented:
> Inner-layer clothing at least should have 100% to spawn.

Underwear have 100% chance to spawn. What other "inner-layer clothing" you are talking about?
Having boots but not having socks? I can think of several cases where soldier could have lost both of his own socks and boots, and then managed to find someone's boots and wear them. Yes, even without socks, right on the bare feet, because I think boots without socks is better than no boots at all.

kevingranade commented:
I think the longarm part of this is reasonable, though no opinion about the numbers.
either the longarm has no sling, so it's gone
or it has a strap and was not worn, so it's gone.
or it has a strap and was worn when the person died, so it's still there.

The clothing and supply loadout is already assuming that they aren't carrying a rucksack, so I don't really follow the argument that random pieces of clothing are going to be missing.

The thing to keep in mind is these soldiers and police would have been deployed for days or maybe weeks, not months and years, so they didn't have time for their loadout to deteriorate to this extent.

Night-Pryanik commented:
> The clothing and supply loadout is already assuming that they aren't carrying a rucksack

Rucksack has a chance to spawn alongside with other clothing, and gear loadout spawn roll isn't tied to the roll of rucksack spawning.

> I don't really follow the argument that random pieces of clothing are going to be missing.

Things happen during the apocalypse, and with the help of imagination one could think of several situation where both live and undead soldiers lost their gear.

Night-Pryanik commented:
I returned guaranteed spawn of underwear such as t-shirts and socks. 

I find helmet or entrenching tool missing from time to time is perfectly reasonable, and don't want to make them spawn always, like they used to, but I could revert my changes to clothing and gear chances to spawn if needed. @kevingranade should I revert them?

ymber commented:
The only thing in here that should have reduced chance to spawn without a rucksack is MREs. I accounted for chances of lost or missing gear when I fixed these. All changes in here except weapon spawn chances are inappropriate and even that introduces a bigger logic hole than the one it addresses.

Night-Pryanik commented:
> All changes in here except weapon spawn chances are inappropriate and even that introduces a bigger logic hole than the one it addresses.

You know, I see here only your statements like "your changes are inappropriate" and other insults. You didn't provide even a single reason of why are they inappropriate besides of your subjective opinion. I even directly asked you about what are you suggesting instead of my change, but you never answered.

FuelType-Memes commented:
> You know, I see here only your statements like "your changes are inappropriate" and other insults. You didn't provide even a single reason of why are they inappropriate besides of your subjective opinion. I even directly asked you about what are you suggesting instead of my change, but you never answered.

I think such a heated response is quite inappropriate. There are multiple people in this PR trying to understand your logic and reasoning, getting riled up is unproductive

In my opinion drops should depend on where we encounter a zombie soldier. You wear different gear in the barracks, in the field or in a lab. But such a change seems bigger than this PR. 

I agree with lowering the chances for guns and gear, maybe even for a couple of clothing items. I think some people might have been caught literally with their pants down

ymber commented:
I accounted for chances to lose gear or have never had it when I made all this. Look at the item group definitions. More than 60% of infantry should have entrenching tools if they've all got their standard issue gear. It's absolutely not reasonable for them to lose a substantial amount of their gear down to basic uniform when this stuff is all worn or carried in pouches designed to not be lost or damaged in combat. The monster descriptions even state that they have their gear. Given that all zombie soldiers have high armor values the spawn chance for ballistic armor should actually be 100%. The rationale for reducing weapon spawns in the monster drops is reasonable but you haven't made them spawn where they would have been dropped and it's a worse problem for them to be completely missing than for them to spawn on zombie soldiers when they should spawn on the ground somewhere. All these changes amount to a regression. I've given reasons why this doesn't make sense and you've ignored them.

Night-Pryanik commented:
> I've given reasons why this doesn't make sense and you've ignored them.

Do show me, where and when exactly you've given me your reasons that I've ignored.

I reverted back all changes to clothing, common infantry gear and MREs spawn chances. Not because I agree with your reasonings (in fact, I don't find them convincing at all), but because I don't want to continue this senseless argument anymore.

> The rationale for reducing weapon spawns in the monster drops is reasonable but you haven't made them spawn where they would have been dropped and it's a worse problem for them to be completely missing than for them to spawn on zombie soldiers when they should spawn on the ground somewhere.

You really think that our current situation where zombies somehow holding weapons (which contradicts our lore) is ok?

davidpwbrown commented:
I mean, you say you dont want to continue the argument but then you ask them to continue it.
Shall we just agree to disagree on all sides here and leave it in the hands of the maintainers to decide what is suitable, i feel you've both outlined your positions pretty well.

( before things get more heated :) )

thethunderhawk commented:
@ymber I agree with you in practice, all soldier zeds should have their equipment mostly accounted for.

However, those things that should not be accounted for in the drop list for a zombie soldier, such as a weapon without a strap, seem to me to be out of the scope of this PR. Zombie soldiers not carrying weapons in their hands is pretty much mandatory. Therefore, abandoned gear is also mandatory. But, as far as I see it, it’s not on @Night-Pryanik to deal with that right now.

kevingranade commented:
I agree that it's not @Night-Pryanik 's "job" to e.g. place replacement guns in map extras or whatever else is needed to balance the change to gun availability, but it's also the case that the shift to gun availability is a potential blocker to merging the PR.

Now how hard a blocker that is is up in the air, I'm not clear on what this actually does to gun availability numbers.

Night-Pryanik commented:
> I mean, you say you dont want to continue the argument but then you ask them to continue it.

I don't want to continue argument about _clothing and gear_ spawn chances, but I'm not against argument about _weapons_ spawn chances.

> Now how hard a blocker that is is up in the air, I'm not clear on what this actually does to gun availability numbers.

https://github.com/CleverRaven/Cataclysm-DDA/pull/36446/files#diff-f05639cd70843dc9eef0931a15cf7491L16-L35
Prior to #36446, long-barreled firearms dropped with only 20% (!) chance, and AFAIK we were totally fine with that. #36446 made these guns' spawn guaranteed, and IMO that's unreasonable. So my changes leading back to decrease in spawn chances to 40% doesn't really affect pre-#36446 gun availability - guns are still available with some chance.

ymber commented:
Increasing gun availability was deliberate. 20% chance for soldiers to have guns was bizarrely low. Cutting drop rates on soldiers makes sense but the other 60% of the weapons need to be accounted for. I think the best solution is to add some nested map chunks to roadblocks and crashed helicopters and such for where they died with blood and stuff they might have been holding and such.

Night-Pryanik commented:
> l_enforcer_45 in sidearms_cop doesn't exist anymore.

Fixed.

> The rifles dropped by bio-operators only have a 0.75 chance to spawn with the SOPMOD contents group and that has a 0.7 chance to spawn a sling so some of those rifles will spawn without slings.

Will be fixed later at some point.

> Less than 40% of soldiers would have been wearing their weapons on slings when they died

Increased chance to spawn guns from 40% to 70%. I hope this will suffice.

> As this currently is it also means no military sniper rifles or DMRs can be found in the field.

Added 10% chance to spawn sniper rifles on corpses of soldiers on military roadblocks. More interesting locations for them could be added later.
